src/main/java/org/owasp/benchmark/score/parsers/ContrastReader.java

```diff
@@ -0,0 +1,152 @@
+				if (line != null) {
+					if (line.startsWith("{\"hash\":")) {
+						parseContrastFinding(tr, line);
+					} else if (line.contains("Agent Version:")) {
+						String version = line.substring(line.indexOf("Version:") + 8);
+						tr.setToolVersion(version.trim());
+					} else if (line.contains("DEBUG - >>> [URL") && line.contains("BenchmarkTest00001")) {
+						firstLine = line;
+					} else if (line.contains("DEBUG - >>> [URL")) {
+						lastLine = line;
+					}
+				}
+			} catch (Exception ex) {
+				ex.printStackTrace();
+			}
+		}
+		reader.close();
+		tr.setTime(calculateTime(firstLine, lastLine));
+		return tr;
+	}
+
+
+    private void parseContrastFinding(TestResults tr, String json) throws Exception {
+        TestCaseResult tcr = new TestCaseResult();
+        
+        try {
+	        JSONObject obj = new JSONObject(json);
+	        String ruleId = obj.getString( "ruleId" );
+	        tcr.setCWE(cweLookup(ruleId));
+	        tcr.setCategory(ruleId);
+	
+	        JSONObject request = obj.getJSONObject("request");	        
+	        String uri = request.getString("uri" );
+        
+	        if ( uri.contains( "BenchmarkTest" ) ) {
+		        String testNumber = uri.substring( uri.lastIndexOf('/') + "BenchmarkTest".length() + 1 );
+	            tcr.setNumber(Integer.parseInt(testNumber));
+		        if (tcr.getCWE() != 0) {
+		            // System.out.println( tcr.getNumber() + "\t" + tcr.getCWE() + "\t" + tcr.getCategory() );
+		            tr.put(tcr);
+		        }
+	        }
+        } catch (Exception e) {
+            // System.err.println("> Parse error: " + json);
+            // e.printStackTrace();
+        }
+    }
+
+    
+	private static int cweLookup(String rule) {
+		switch (rule) {
+		case "cookie-flags-missing":
+			return 614; // insecure cookie use
+		case "sql-injection":
+			return 89; // sql injection
+		case "cmd-injection":
+			return 78; // command injection
+		case "ldap-injection":
+			return 90; // ldap injection
+		case "header-injection":
+			return 113; // header injection
+		case "hql-injection":
+			return 564; // hql injection
+		case "unsafe-readline":
+			return 0000; // unsafe readline
+		case "reflection-injection":
+			return 0000; // reflection injection
+		case "reflected-xss":
+			return 79; // xss
+		case "xpath-injection":
+			return 643; // xpath injection
+		case "path-traversal":
+			return 22; // path traversal
+		case "crypto-bad-mac":
+			return 328; // weak hash
+		case "crypto-weak-randomness":
+			return 330; // weak random
+		case "crypto-bad-ciphers":
+			return 327; // weak encryption
+		case "trust-boundary-violation":
+			return 501; // trust boundary
+		case "xxe":
+			return 611; // xml entity
+		}
+		return 0;
+	}
+
+	private String calculateTime(String firstLine, String lastLine) {
+		try {
+			String start = firstLine.split(" ")[1];
+			String stop = lastLine.split(" ")[1];
+			SimpleDateFormat sdf = new SimpleDateFormat("HH:mm:ss,SSS");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.